c_sharp_json_diff/JsonDiff.cs

```diff
@@ -1,55 +1,63 @@
         {
-           ICollection<String> firstKeys = first.Keys;
-            ICollection<String> secondKeys = second.Keys;
-            List<String> matches = new List<string>();
-            Result result = new Result();
-            foreach (string key in secondKeys)
-            {
-                if (firstKeys.Contains(key))
-                {
-                    matches.Add(key);
-                }
-            }
-            foreach(string key in matches)
+            string diff = JsonDiffBetweenTwo(left, right);
+            JObject diffJsonObject = JObject.Parse(diff);
+            foreach(KeyValuePair<string, JToken?> prop in diffJsonObject)
             {
-                first.TryGetValue(key, out double firstValue);
-                second.TryGetValue(key, out double secondValue);
-                if(firstValue != secondValue)
+                if(prop.Value != null)
                 {
-                    Diff item = new Diff();
-                    item.jsonProperty = key;
-                    item.previous = firstValue;
-                    item.current = secondValue;
-                    item.difference = Math.Abs(firstValue - secondValue);
-                    result.Diffs.Add(item);
+                    JArray valuesAsArray = (JArray)prop.Value;
+                    if (valuesAsArray[0] != null && valuesAsArray[1] != null)
+                    {
+                        string leftValueAsString = valuesAsArray[0].ToString();
+                        string rightValueAsString = valuesAsArray[1].ToString();
+                        if (int.TryParse(leftValueAsString, out int leftValueAsInt) && int.TryParse(rightValueAsString, out int rightValueAsInt))
+                        {
+                            int diffValue = Math.Abs(leftValueAsInt - rightValueAsInt);
+                            valuesAsArray.Add(diffValue);
+                        }
+                        else if (float.TryParse(leftValueAsString, out float leftValueAsFloat) && float.TryParse(rightValueAsString, out float rightValueAsFloat))
+                        {
+                            float diffValue = Math.Abs(leftValueAsFloat - rightValueAsFloat);
+                            valuesAsArray.Add(diffValue);
+                        }
+                        else if (double.TryParse(leftValueAsString, out double leftValueAsDouble) && double.TryParse(rightValueAsString, out double rightValueAsDouble))
+                        {
+                            double diffValue = Math.Abs(leftValueAsDouble - rightValueAsDouble);
+                            valuesAsArray.Add(diffValue);
+                        }
+                        else
+                        {
+                            prop.Value.Remove();
+                        }
+                    }
                 }
             }
-            return result;
+            string result = diffJsonObject.ToString(Formatting.Indented);
```

Comment:
Remove this unnecessary code

---
**Sema Reaction:** :hammer_and_wrench: This code needs a fix